### PR TITLE
Adds the keys() function to Cypher

### PR DESCRIPTION
**Pull request by systay**
Example : `MATCH (n) RETURN keys(n)`

### CODE REVIEW

File: community/cypher/docs/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/FunctionsTest.scala [hunk 1]
```diff
@@ -147,6 +147,22 @@ class FunctionsTest extends DocumentingTestBase {
     )
   }
 
+  @Test def keys() {
+    testThis(
+      title = "KEYS",
+      syntax = "KEYS( node )",
```
Comment (jotomo) [inline community/cypher/docs/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/FunctionsTest.scala]: Seems this bit was missed when _keys()_ was expanded to work for relationships as well?

File: community/cypher/cypher-compiler-2.2/src/main/scala/org/neo4j/cypher/internal/compiler/v2_2/commands/expressions/KeysFunction.scala [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      queryCtx.getPropertiesForNode(n.getId).map { case (v) =>
+        queryCtx.getPropertyKeyName(v.toInt)
+      }.toList
+    case rel: Relationship =>
+      val queryCtx: QueryContext = state.query
+      queryCtx.getPropertiesForRelationship(rel.getId).map { case (v) =>
+        queryCtx.getPropertyKeyName(v.toInt)
+      }.toList
+    case null =>  null
+    case _    =>  throw new CypherTypeException("keys() expected a Node or Relationship but it was called with something else")
```
Comment (pontusmelke) [inline community/cypher/cypher-compiler-2.2/src/main/scala/org/neo4j/cypher/internal/compiler/v2_2/commands/expressions/KeysFunction.scala]: Can't we do something similar as in `compute` above, and not just saying 'called with something else'

File: community/cypher/cypher-compiler-2.2/src/main/scala/org/neo4j/cypher/internal/compiler/v2_2/commands/expressions/KeysFunction.scala [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+  def arguments = Seq(nodeOrRelationshipExpr)
+
+  def symbolTableDependencies = nodeOrRelationshipExpr.symbolTableDependencies
+
+  protected def calculateType(symbols: SymbolTable) = {
+    CTCollection(CTString)
+  }
+
+  override def localEffects = Effects.READS_NODES
```
Comment (pontusmelke) [inline community/cypher/cypher-compiler-2.2/src/main/scala/org/neo4j/cypher/internal/compiler/v2_2/commands/expressions/KeysFunction.scala]: READS_RELATIONSHIPS sometimes as well?

File: community/cypher/cypher-compiler-2.2/src/test/scala/org/neo4j/cypher/internal/compiler/v2_2/commands/expressions/KeysFunctionTest.scala [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+import org.mockito.stubbing.Answer
+import org.neo4j.cypher.internal.commons.CypherFunSuite
+import org.neo4j.cypher.internal.compiler.v2_2._
+import org.neo4j.cypher.internal.compiler.v2_2.pipes.QueryStateHelper
+import org.neo4j.cypher.internal.compiler.v2_2.spi.QueryContext
+import org.neo4j.graphdb.Node
+
+class KeysFunctionTest extends CypherFunSuite {
+
+  test("testPropertyKeys") {
```
Comment (pontusmelke) [inline community/cypher/cypher-compiler-2.2/src/test/scala/org/neo4j/cypher/internal/compiler/v2_2/commands/expressions/KeysFunctionTest.scala]: spaces are allowed:)

File: community/cypher/cypher/src/test/scala/org/neo4j/cypher/KeysAcceptanceTest.scala [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+package org.neo4j.cypher
+
+import internal.helpers.CollectionSupport
+import org.scalatest.Assertions
+import org.neo4j.graphdb.{Relationship, Node}
+import org.scalautils.LegacyTripleEquals
+
+class KeysAcceptanceTest extends ExecutionEngineFunSuite  with QueryStatisticsTestSupport {
+
+  test("Using_keys_function_with_NODE_Not_Empty_result") {
```
Comment (pontusmelke) [inline community/cypher/cypher/src/test/scala/org/neo4j/cypher/KeysAcceptanceTest.scala]: spaces are allowed here as well

**Comment by systay** [comment]
Re-opened by @antonio-castellon here #3943

**Pull request closed without merging**